src/plugins/intel_cpu/src/nodes/kernels/stack_allocator.hpp

```diff
@@ -0,0 +1,474 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <ie/ie_common.h>
+#include <cpu/x64/jit_generator.hpp>
+
+namespace ov {
+namespace intel_cpu {
+
+class StackAllocator final {
+public:
+    using jit_generator = dnnl::impl::cpu::x64::jit_generator;
+    class Transaction;
+    class Address;
+    template<typename TReg>
+    class RegAddress;
```

Comment:
I have a time to comment on this renaming ...

As for me this name `RegAddress` raise more questions ...
What does it mean to have address of register ?

This renaming provide inconsistency with `RegisterPool::Reg<...>`
With previous name it was possible to write the general helper function like this:
```cpp
template <typename TAllocator, typename TReg>
void custom_gather(TAllocator::Reg<TReg>& arg0, ...) {
   ...
}
```
But with such naming convention it is not possible ...
Anyway as for me it is enough to have prefix before class name that describe where it was allocated

Reply:
> What does it mean to have address of register ?

This means the address of a data block of a certain register type size.
> With previous name it was possible to write the general helper function like this:

The problem is that the `StackAllocator::Reg` is not the same entity as `RegisterPool::Reg` since it is not a hardware register, but an memory address, with completely different access metrics (latency could be equal to DRAM access time in a worst case), so it should not be used interchangeably by default, but only intentionally when the developer understand all the drawbacks. So such generic templated code may lead to unexpected performance drop due to memory access instead of register, but it would not be obvious from the source code. That is why, at least in my understanding, we should highlight this difference using slightly different name.

Reply:
I think `StackAllocator::` prefix is enough.
Anyway as for me this renaming is very subjective ...